pallets/crowdloans/src/lib.rs

```diff
@@ -959,6 +976,118 @@ pub mod pallet {
             }
             Ok(().into())
         }
+
+        // ********************************************************************************************************
+        #[pallet::weight(<T as Config>::WeightInfo::refund())]
+        #[transactional]
+        pub fn refund(origin: OriginFor<T>, crowdloan: ParaId) -> DispatchResult {
+
+            let who = ensure_signed(origin)?;
+
+            let mut vault = Self::current_vault(crowdloan).ok_or(Error::<T>::VaultDoesNotExist)?;
+            
+            // 1. check phase, should be Closed or Failed
+            ensure!(
+                vault.phase == VaultPhase::Closed || vault.phase == VaultPhase::Failed,
+                Error::<T>::IncorrectVaultPhase
```

Comment:
refund is for all contributors, it's a for loop which will do `RemoveKeysLimit` every time

Reply:
Should this be happening outside it ?
```
ensure!(
                vault.phase == VaultPhase::Closed || vault.phase == VaultPhase::Failed,
                Error::<T>::IncorrectVaultPhase```

Reply:
no it's ok to put here